src/jsm.c

```diff
@@ -1718,6 +1718,75 @@ js_EraseMsg(jsCtx *js, const char *stream, uint64_t seq, jsOptions *opts, jsErrC
 // Account related functions
 //
 
+static natsStatus
+_unmarshalAccLimits(nats_JSON *json, jsAccountLimits *limits)
+{
+    natsStatus  s;
+    nats_JSON   *obj = NULL;
+
+    s = nats_JSONGetObject(json, "limits", &obj);
+    if (obj == NULL)
+        return NATS_UPDATE_ERR_STACK(s);
+
+    IFOK(s, nats_JSONGetLong(obj, "max_memory", &(limits->MaxMemory)));
+    IFOK(s, nats_JSONGetLong(obj, "max_storage", &(limits->MaxStore)));
+    IFOK(s, nats_JSONGetLong(obj, "max_streams", &(limits->MaxStreams)));
+    IFOK(s, nats_JSONGetLong(obj, "max_consumers", &(limits->MaxConsumers)));
+    IFOK(s, nats_JSONGetLong(obj, "max_ack_pending", &(limits->MaxAckPending)));
+    IFOK(s, nats_JSONGetLong(obj, "memory_max_stream_bytes", &(limits->MemoryMaxStreamBytes)));
+    IFOK(s, nats_JSONGetLong(obj, "storage_max_stream_bytes", &(limits->StoreMaxStreamBytes)));
+    IFOK(s, nats_JSONGetBool(obj, "max_bytes_required", &(limits->MaxBytesRequired)));
+
+    return NATS_UPDATE_ERR_STACK(s);
+}
+
+static natsStatus
+_fillTier(void *userInfo, const char *subject, nats_JSONField *f)
+{
+    jsAccountInfo   *ai     = (jsAccountInfo*) userInfo;
+    natsStatus      s       = NATS_OK;
+    jsTier          *t      = NULL;
+    nats_JSON       *json   = f->value.vobj;
+
+    t = (jsTier*) NATS_CALLOC(1, sizeof(jsTier));
+    if (t == NULL)
+        return nats_setDefaultError(NATS_NO_MEMORY);
+
+    ai->Tiers[ai->TiersLen++] = t;
+
+    DUP_STRING(s, t->Name, subject);
+    IFOK(s, nats_JSONGetULong(json, "memory", &(t->Memory)));
```

Comment:
For these you can get all the same values from `_unmarshalAccLimits`

Reply:
No, `_unmarshalAccLimits` deals with "max" limits. The "memory", etc.. fields are also parsed in top-level of js_unmarshalAccountInfo() and so could have made that a function, but not worth it.

Note that in this function (bottom) I do call `_unmarshalAccLimits` for the account limits.

Reply:
That is what I thought but wanted to verify.